crates/ra_hir/src/semantics.rs

```diff
@@ -271,6 +272,86 @@ impl<'db, DB: HirDatabase> Semantics<'db, DB> {
     pub fn assert_contains_node(&self, node: &SyntaxNode) {
         self.imp.assert_contains_node(node)
     }
+
+    pub fn is_unsafe_method_call(&self, method_call_expr: ast::MethodCallExpr) -> Option<()> {
```

Comment:
Semantics is a facade, boundary API, so it's important to exercise care when touching its public functions.

`pub fn is_xxx(...) -> Option<()>` as an API does not make sense. Like, it works, but I hope it's obvious that the `-> Option<()>` signature is just wrong for public `is_` function (unless there are some **very** specific circumstances). 

Additionally, note that all methods of `Semantics` follow a certain pattern, and these new additions deviate from the pattern (see git blame for the reasoning). 


I am more nitpicky than usual with this PR. This is because the code itself is fine, and I feel that slightly correcting certain underling *values* would help with improving code in general (as opposed to fixing some specific problems) :0)

Reply:
No worries, I'm in no rush to get this out. I would rather ship something clean than broken, and with all the rebases, I sometimes forget to go back and clean things up. Thanks for being so diligent and patient :)